src/System.Collections.Concurrent/tests/ConcurrentDicionary/ConcurrentDictionaryTests.cs

```diff
@@ -441,25 +444,47 @@ private static void TestGetOrAddOrUpdate(int cLevel, int initSize, int threads,
                             {
                                 if (isAdd)
                                 {
-                                    //call either of the two overloads of GetOrAdd
-                                    if (j + ii % 2 == 0)
-                                    {
-                                        dict.GetOrAdd(j, -j);
-                                    }
-                                    else
+                                    //call one of the overloads of GetOrAdd
+                                    switch (j % 3)
                                     {
-                                        dict.GetOrAdd(j, x => -x);
+                                        case 0:
+                                            dict.GetOrAdd(j, -j);
+                                            break;
+                                        case 1:
+                                            dict.GetOrAdd(j, x => -x);
+                                            break;
+                                        case 2:
+                                            if (s_getOrAddNetCoreApp11 != null)
+                                            {
+                                                s_getOrAddNetCoreApp11(dict, j);
```

Comment:
An alternative would be to implement the new method behavior as extension methods in the test that are compiled in only when testing the older version, and then leave this test itself clean.